### ISSUE TITLE
 Asynchronous use of GM.getValue returns the previous value.

### ISSUE THREAD

**Issue by Shi-553** [quoid/userscripts#220]
iPadOS15.0.2
Userscripts 1.0.2
This problem is not related to script injection.
The test script was executed successfully.
```js
// ==UserScript==
// @name         TEST
// @description  TEST
// @match        *://*/*
// @grant        GM.getValue
// @grant        GM.setValue
// ==/UserScript==

const A = async () => {
    console.log("AAA", await GM.getValue("AAA"));
}
const B = async () => {
    console.log("BBB", await GM.getValue("BBB"));
}

(async () => {
    await GM.setValue("AAA", "AAAAA");
    await GM.setValue("BBB", "BBBBB");

    A();
    B();
    B();
    await B();
    await B();
})()
```

Run the script above.

Expected behavior: Only "AAA AAAAAA" or "BBB BBBBBB" will be output.
Actual behavior: Sometimes "BBB AAAAAA" is output.

I tried it with Tampermonkey on my PC and it works as expected.

**Comment by quoid** [comment]
@Shi-553 Thanks for the report, I did notice this kind of behavior in the past. I believe the promises need a unique id to ensure their callbacks do not respond to the incorrect window messages (which is what I believe is happening right now).

I am testing a version with a fix on my machine right now and it will be applied to a future version. I might release the fix as a patch version soon.

**Comment by quoid** [comment]
I have not yet experienced the issue you described **with the fix applied**. I will keep testing.

<img width="1234" alt="Screen Shot 2022-01-24 at 9 27 11 AM" src="https://user-images.githubusercontent.com/7660254/150802694-6256b4ad-3208-450a-a494-ed9e50d67ffb.png">

**Comment by quoid** [comment]
There's a few other API bugs I want to work out, I am think this fix might be rolled into 4.1

https://github.com/quoid/userscripts/issues/221

**Comment by quoid** [comment]
done in [v4.1.0, v1.1.0](https://github.com/quoid/userscripts/releases/tag/v4.1.0)
